public/app/features/query/components/QueryEditorRowHeader.test.tsx

```diff
@@ -13,16 +14,29 @@ const mockDS = mockDataSource({
   type: DataSourceType.Alertmanager,
 });
 
+const mockVariable = mockDataSource({
+  name: '${dsVariable}',
+  type: 'datasource',
+});
+
 jest.mock('@grafana/runtime/src/services/dataSourceSrv', () => {
   return {
     getDataSourceSrv: () => ({
       get: () => Promise.resolve(mockDS),
-      getList: () => [mockDS],
+      getList: ({ variables }: { variables: boolean }) => (variables ? [mockDS, mockVariable] : [mockDS]),
       getInstanceSettings: () => mockDS,
     }),
   };
 });
 
+jest.mock('@grafana/runtime/src/services/templateSrv', () => {
+  return {
+    getTemplateSrv: () => ({
+      getVariables: () => [mockVariable],
+    }),
+  };
+});
+
```

Comment:
Do we need this mock anymore? I think by mocking the `getList` we will never get to a point where `templateSrv` is used

Reply:
good spot @JoaoSilvaGrafana 🙌